src/diffcalc_API/services/hkl.py

```diff
@@ -22,33 +21,39 @@ async def lab_position_from_miller_indices(
     if all([idx == 0 for idx in miller_indices]):
         raise InvalidMillerIndicesError()
 
-    all_positions = hklcalc.get_position(*miller_indices, wavelength)
+    all_positions = hklcalc.get_position(*miller_indices.dict().values(), wavelength)
 
     return combine_lab_position_results(all_positions)
 
 
 async def miller_indices_from_lab_position(
     name: str,
-    pos: Tuple[float, float, float, float, float, float],
+    pos: PositionModel,
     wavelength: float,
     store: HklCalcStore,
     collection: Optional[str],
-) -> Tuple[Any, ...]:
+) -> HklModel:
     hklcalc = await store.load(name, collection)
-    position = hklcalc.get_hkl(Position(*pos), wavelength)
-    return tuple(np.round(position, 16))
+    hkl = np.round(hklcalc.get_hkl(Position(**pos.dict()), wavelength), 16)
+    return HklModel(h=hkl[0], k=hkl[1], l=hkl[2])
 
 
 async def scan_hkl(
     name: str,
-    start: PositionType,
-    stop: PositionType,
-    inc: PositionType,
+    start: List[float],
+    stop: List[float],
+    inc: List[float],
```

Comment:
These may be better typed as `tuple[float, float, float]`

Reply:
mypy seems to complain when I do this, and wants me to use Tuple[float, float, float], which I shouldn't use because currently fastAPI doesn't support this.